index.js

```diff
@@ -58,10 +58,18 @@ class TemplateCompiler extends Filter {
   }
 
   processString(string, relativePath) {
-    return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
-      contents: string,
-      moduleName: relativePath
-    }) + ';';
+    try {
+      return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
+        contents: string,
+        moduleName: relativePath
+      }) + ';';
+    } catch(error) {
+      // augment with location and type information and re-throw.
+      error.type = 'Template Compiler Error';
+      error.location = error.location.start;
```

Comment:
Error might not be an object and even if it is it may not have location. This catch clause will likely need to be slightly more resilient